Untitled Game/src/Levels/LevelWithRoomsMockup.gd

```diff
@@ -0,0 +1,80 @@
+extends Node2D
+
+var _LABEL_POSITION_OFFSET = 100
+
+var _lockCamera = false
+var _startCountDown = 5
+var _timerOn = false
+
+onready var camera = $Camera2D
+onready var player = $LirikYaki
+onready var firstLockArea = $FirstRoomLockArea
+onready var cameraManager = CameraManager.new(player, camera)
+onready var label = $CanvasLayer/Label
+onready var leftFirstRoomCollider = $FirstRoomBattleBoundary/LeftSide
+onready var rightFirstRoomCollider = $FirstRoomBattleBoundary/RightSide
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	cameraManager.followPlayerIfUnlocked()
+	_handleTimerUpdate(delta)
+
+
+# Event when hitting the First Room lock area, will lock camera, and player to that area
+func _on_FirstRoomLockArea_area_entered(area):
+	print("first collision detect: " + area.name)
+	# I use players position to lock the camera to, but it might be better to define
+	# an actual space on the map/scene
+	cameraManager.lockCameraToPosition(player.position)
+	firstLockArea.queue_free() # disable any recurring events
+	enableFirstRoomBoundaries()
+	_startTimer()
+
```

Comment:
instead of making a method for every room, lets try to make this dynamic so that we can attach it to multiple areas more easily when we have a lot more.

Reply:
Fixed in latest commit 9917d3fc3008aaa25ad3eb7ba8ba27cec4a1f1a2.